### ISSUE TITLE
Refactor StoragesConfigurationController to use Form Validation and make sure it follows the How-To-Write-Controllers guide

### ISSUE THREAD

**Issue by steve-todorov** [strongbox/strongbox#695]
The `StoragesConfigurationController` needs to be refactored so that it uses spring form validation. Also, make sure that the controller returns the proper responses according to the [How-To-Implement-Spring-Controllers](https://github.com/strongbox/strongbox/wiki/How-To-Implement-Spring-Controllers) when there is an error in the validation. 

Validation rules for `Storage`:

| Field | Validation  |
| --- | --- |
| id  | required, not blank, string  |
| basedir  | optional |
| repositories | optional |

Validation rules for `Repository`

| Field | Validation |
| --- | --- |
| id  | required, not blank |
| type | required, not blank, possible values - "hosted","proxy", "group" |
| basedir | optional | 
| policy | required, not blank, value can be `release`, `snapshot` or `mixed`  (create constants in `RepositoryPolicyEnum`) |
| implementation | required, not blank, accepts only `file-system` for now |
| layout | required, not blank, value needs to be a valid `layout` **TODO** make a service which returns the layouts list.If you'd like, you can make a separate PR for this before starting on this task |
| secured | optional, defaults to false |
| status | required, not blank, value needs to be a valid `status` **TODO** make a service which returns the status list. If you'd like, you can make a separate PR for this before starting on this task |
| artifactMaxSize | optional, defaults to `0` |
| trashEnabled | optional, defaults to `true` |
| allowsForceDeletion | optional, defaults to `false` |
| allowsDeployment | optional, defaults to `true` |
| allowsRedeployment | optional, defaults to `false` |
| allowsDelete | optional, defaults to `true` |
| allowsDirectoryBrowsing | optional, defaults to `true` |
| checksumHeadersEnabled | optional, defaults to `false` |
| proxyConfiguration | optional, should use same validation as [#681](https://github.com/strongbox/strongbox/issues/681) (preferably same form) |
| remoteRepository | optional, check `Validation rules for RemoteRepository` |
| httpConnectionPool | optional, if set must be an integer |
| groupRepositories | optional | 
| artifactCoordinateValidators | optional |

Validation rules for `RemoteRepository`

| Field | Validation |
| --- | --- |
| url  | required, not blank |
| downloadRemoteIndexes | optional, defaults to `false` | 
| autoBlocking | optional, defaults to `false` |
| checksumValidation | optional, defaults to `false` |
| username | optional |
| password | optional |
| checksumPolicy | optional |
| checkIntervalSeconds | required, not blank |
| allowsDirectoryBrowsing | optional, defaults to `true` |
| autoImportRemoteSSLCertificate | optional, defaults to `false` |

**Comment by ptirador** [comment]
Hello, @steve-todorov. As you state in the description, I'm starting with these tasks:
- Make a service which returns the layouts list => How can I retrieve this?
- Make a service which returns the status list => Given the status as an enum type (`RepositoryStatusEnum`), should I just return an array of enums (`RepositoryStatusEnum.values()`), or their string values?

**Comment by carlspring** [comment]
> - Make a service which returns the layouts list => How can I retrieve this?

Check the `LayoutProvidersRegistry`. In fact, I think you should be able to use that directly.

> - Make a service which returns the status list => Given the status as an enum type (`RepositoryStatusEnum`), should I just return an array of enums (`RepositoryStatusEnum.values()`), or their string values?

This should be their string values.

**Comment by steve-todorov** [comment]
@ptirador how are things going? Did you had a chance to start with this yet? :)

**Comment by ptirador** [comment]
Hi! I've started, but I haven't had much time lately. I'll keep going in a few days ;)

**Comment by carlspring** [comment]
@ptirador ,

@fuss86 has started an urgent refactoring that will affect you. This should take a few days, hopefully not longer than 30/05/2018. If you're up for tasks during this time, could you please ping me, or pick something else?

Sorry for the inconvenience!

**Comment by carlspring** [comment]
@fuss86 : Can the work on this be resumed/continued?

**Comment by fuss86** [comment]
Yes, please proceed @ptirador. Please rebase to the latest master, there was a lot of changes in this area.

**Comment by steve-todorov** [comment]
@ptirador are you still interested in doing this task? :)

**Comment by ptirador** [comment]
I'm not having much time lately, so please proceed to assign it to anyone else.

**Comment by steve-todorov** [comment]
No worries, many thanks for replying! :)

**Comment by fuss86** [comment]
I'm picking up this task :)

**Comment by fuss86** [comment]
> Validation rules for Repository

there is missed a `type` with valid values  `"hosted","proxy", "group"`

**Comment by fuss86** [comment]
in progress  #765

**Comment by steve-todorov** [comment]
> there is missed a type with valid values "hosted","proxy", "group"

Good catch! I've updated the description. :)
